src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/on_app_interface_unregistered_notification.cc

```diff
@@ -57,7 +57,17 @@ OnAppInterfaceUnregisteredNotification::
 void OnAppInterfaceUnregisteredNotification::Run() {
   LOG4CXX_AUTO_TRACE(logger_);
 
-  SendNotification();
+  const auto app_id = static_cast<uint32_t>(
+      (*message_)[strings::params][strings::connection_key].asUInt());
+  const auto app = application_manager_.application(app_id);
+
+  using Reason = mobile_apis::AppInterfaceUnregisteredReason::eType;
+  const auto reason = static_cast<Reason>(
+      (*message_)[strings::msg_params][strings::reason].asInt());
+  const auto final_message = app && app->webengine_projection_enabled() &&
+                             Reason::RESOURCE_CONSTRAINT == reason;
```

Comment:
@mvorobio Move this code please to a separate function: 
```cpp
bool IsFinalMessage() {
  const auto app_id = static_cast<uint32_t>(
      (*message_)[strings::params][strings::connection_key].asUInt());
  const auto app = application_manager_.application(app_id);

  using Reason = mobile_apis::AppInterfaceUnregisteredReason::eType;
  const auto reason = static_cast<Reason>(
      (*message_)[strings::msg_params][strings::reason].asInt());
  return app && app->webengine_projection_enabled() &&
                             Reason::RESOURCE_CONSTRAINT == reason;
}

void OnAppInterfaceUnregisteredNotification::Run() {
  LOG4CXX_AUTO_TRACE(logger_);
  const bool final_message = IsFinalMessage();
  SendNotification(final_message);
}
```

Reply:
@AByzhynar please check https://github.com/LuxoftSDL/sdl_core/pull/18/commits/7bc1f2ce4c04d80778a833d7022995d806ce5788